scripts/exaqm_make_ics.sh

```diff
@@ -13,583 +13,13 @@ export pgm=aqm_make_ics
-  fi
-
-  if [ -n "${file_names:-}" ] ; then
-    additional_flags="$additional_flags --file_templates ${file_names[@]}"
-  fi
-
-  if [ -n "${input_file_path:-}" ] ; then
-    data_stores="disk $data_stores"
-    additional_flags="$additional_flags --input_file_path ${input_file_path}"
-  fi
-
-  if [ $SYMLINK_FIX_FILES = "TRUE" ]; then
-    additional_flags="$additional_flags --symlink"
-  fi
-
-  if [ $DO_ENSEMBLE == "TRUE" ] ; then
-    mem_dir="/mem{mem:03d}"
-    member_list=(1 ${NUM_ENS_MEMBERS})
-    additional_flags="$additional_flags --members ${member_list[@]}"
-  fi
-
-  EXTRN_DEFNS="${NET}.${cycle}.${EXTRN_MDL_NAME}.ICS.${EXTRN_MDL_VAR_DEFNS_FN}.sh"
-
-  cmd="
-  ${USHaqm}/retrieve_data.py \
-  --debug \
-  --symlink \
-  --file_set ${file_set} \
-  --config ${PARMdir}/data_locations.yml \
-  --cycle_date ${EXTRN_MDL_CDATE} \
-  --data_stores ${data_stores} \
-  --external_model ${EXTRN_MDL_NAME} \
-  --fcst_hrs ${fcst_hrs[@]} \
-  --ics_or_lbcs "ICS" \
-  --output_path ${EXTRN_MDL_STAGING_DIR}${mem_dir} \
-  --summary_file ${EXTRN_DEFNS} \
-  $additional_flags"
-
-  $cmd
-  export err=$?
-
-  if [ $err -ne 0 ]; then
-    message_txt="FATAL ERROR Call to retrieve_data.py failed with a non-zero exit status.
-The command was:
-${cmd}
-"
-    err_exit "${message_txt}"
-  fi
-fi
-#
-#-----------------------------------------------------------------------
-#
-# Source the file containing definitions of variables associated with the
-# external model for ICs.
-#
-#-----------------------------------------------------------------------
-#
-if [ "${RUN_TASK_GET_EXTRN_ICS}" = "FALSE" ]; then
-   extrn_mdl_staging_dir="${DATA}"
-else
-   if [ "${WORKLFOW_MANAGER}" = "ecflow" ]; then  
-     extrn_mdl_staging_dir="${DATAROOT}/${RUN}_get_extrn_ics_${cyc}.${share_pid}"
-     if [ ! -d ${extrn_mdl_staging_dir} ]; then
-       message_txt="FATAL ERROR ${extrn_mdl_staging_dir} not found in production mode"
-       err_exit "${message_txt}"
-     fi
-   else
-      extrn_mdl_staging_dir="${DATAROOT}/get_extrn_ics.${share_pid}"
-   fi
-fi
- extrn_mdl_var_defns_fp="${extrn_mdl_staging_dir}/${NET}.${cycle}.${EXTRN_MDL_NAME_ICS}.ICS.${EXTRN_MDL_VAR_DEFNS_FN}.sh"
-. ${extrn_mdl_var_defns_fp}
-#
-#-----------------------------------------------------------------------
-#
-# Set physics-suite-dependent variable mapping table needed in the FORTRAN
-# namelist file that the chgres_cube executable will read in.
-#
-#-----------------------------------------------------------------------
-#
-varmap_file=""
-
-case "${CCPP_PHYS_SUITE}" in
-#
-  "FV3_GFS_2017_gfdlmp" | \
-  "FV3_GFS_2017_gfdlmp_regional" | \
-  "FV3_GFS_v16" | \
-  "FV3_GFS_v15p2" )
-    varmap_file="GFSphys_var_map.txt"
-    ;;
-  *)
-    message_txt="FATAL ERROR The variable \"varmap_file\" has not yet been specified for 
-this physics suite (CCPP_PHYS_SUITE):
-  CCPP_PHYS_SUITE = \"${CCPP_PHYS_SUITE}\""
-     err_exit "${message_txt}"
-    ;;
-#
-esac
```

Comment:
The appropriate values from this case statement and the next one will already be defined in the experiment configuration YAML because we know enough information at configuration time to pre-determine (harden) this for run-time.